src/components/spaces/canvas/index.js

```diff
@@ -86,15 +85,31 @@ export default class CanvasSpace extends Component{
     this.props.dispatch(changeSelect(next))
   }
 
-  renderMetric(metric) {
+  _selectedMetric() {
+   const {selected} = this.props
+   const metrics = _.get(this.props.denormalizedSpace, 'metrics')
+
+   const item = metrics && metrics.filter(i => _.isEqual(i.location, selected))[0];
+   return item
+  }
+
+  _isAnalysisView(props = this.props) {
```

Comment:
Why does this take a parameter? Just reference this.props inside the function body.